### ISSUE TITLE
`v:iterator.explode` returns exactly the value of `content` if `as` is given

### ISSUE THREAD

**Issue by mbonaker** [FluidTYPO3/vhs#1307]
#32 has been closed in 2013 and locked today when I tried to revive it. It appears to me though that `v:iterator.explode` still (or again) behaves weirdly if used with `as`.

In the mentioned issue @NamelessCoder wrote that this would be the proper usage:

    <v:iterator.explode content="{var}" as="othervariable">
        <f:for each="{othervariable}" as="..."></f:for>
    </v:iterator.explode>

Which expands to this imho, if we fill out the blanks:

    <v:iterator.explode content="red,green,blue" glue="," as="othervariable">
        <f:for each="{othervariable}" as="iter">
            <pre>{iter}</pre>
        </f:for>
    </v:iterator.explode>

Apart from the white spaces the above should generate exactly the following code if I understand this correctly:

    <pre>red</pre>
    <pre>green</pre>
    <pre>blue</pre>

But in fact it gives me:

    red,green,blue

A maybe simpler inline example would be this:

    {iter -> v:iterator.explode(content: 'red,green,blue', glue: ',', as: 'iter') -> v:format.json.encode()}

Which also generates `red,green,blue` instead of `["red", "green", "blue"]`

-----

I have no experience with the code of fluid but I have dug into it a little bit. I have found the connection is this:
`ExplodeViewHelper` uses the trait `CompileWithContentArgumentAndRenderStatic`. That defines `render` where basically the `$renderChildrenClosure` parameter of the `ExplodeViewHelper::renderStatic` method seems to be set to the current value of the `content` argument. The `ExplodeViewHelper` makes a few nested calls but in the end it is assumed that `$renderChildrenClosure` would be the rendered content of the tag which is not true due to the `CompileWithContentArgumentAndRenderStatic`.

So basically the real children of the tag are always completely ignored if the `content` argument is given. But since the `as` tag only makes even any sense with *both*, the `content` argument and the children, this is imho to be considered a bug regarding the `as` argument.

**Comment by NamelessCoder** [comment]
Your investigation is correct, and the second example you give should indeed result in the list of HTML tags you expect. The reason is exactly the content argument vs. the render children closure.

The solution is described in #1254 and we'd of course appreciate pull requests to fix this (there are other ViewHelpers in VHS which use an `as` argument and content argument achieved with the vanilla `CompileWithRenderStatic` trait).

PS: `{varWithCSV -> v:iterator.explode() -> v:format.json.encode()}` also works.

**Comment by mbonaker** [comment]
Well, that issue you mentioned (#1254) exactly describes the whole thing. I should have found that instead of opening a new issue - sorry about that. I'm gonna close this one as duplicate then in favor of the older one.
